### ISSUE TITLE
Devpod watch.sh location?

### ISSUE THREAD

**Issue by sdoxsee** [jenkins-x/jx#1147]
I used 'jx create spring' and have a git repository and everything else. However, when i do 'jx create devpod --sync --reuse' I get into the pod but there is no watch.sh script that builds my image to test (as per https://jenkins-x.io/developing/devpods/#example-workflow-with-an-existing-project). At what point does the watch.sh appear or is that not part of the spring project and only quickstart? Thanks!

Update: Just found more [documentation](https://jenkins-x.io/developing/devpods/#incremental-building)

> So once the skaffold dev -p dev (what watch.sh does) 

Now I can see it built and I can go to the url. (NOTE: still not sure why there was no watch.sh...perhaps this was removed and the documentation needs updating)

However, it still doesn't seem to be watching my changes. Maybe the error below is why
```
skaffold dev -p dev
Starting build...
Sending build context to Docker daemon  17.49MB
Step 1/8 : FROM openjdk:8-jdk-slim
 ---> 954f983e50de
Step 2/8 : ENV PORT 8080
 ---> Using cache
 ---> 5d31b06d4fdd
Step 3/8 : ENV CLASSPATH /opt/lib
 ---> Using cache
 ---> 00396980b264
Step 4/8 : EXPOSE 8080
 ---> Using cache
 ---> 395efbf6caf0
Step 5/8 : COPY pom.xml target/lib* /opt/lib/
 ---> Using cache
 ---> 6448ae323e2b
Step 6/8 : COPY target/*.jar /opt/app.jar
 ---> Using cache
 ---> 40ae182925c2
Step 7/8 : WORKDIR /opt
 ---> Using cache
 ---> 7002287f565d
Step 8/8 : CMD java -jar app.jar
 ---> Using cache
 ---> ca20213eb1af
Successfully built ca20213eb1af
WARN[0000] run: build: build step: tagging image: Error parsing reference: "<no value>/sdoxsee/demo:ca20213eb1af650466af7c43345d9de579b992b368fbf4543bef0c4067552dea" is not a valid repository/tag: invalid reference format
Watching for changes...
```

must be my DOCKER_REGISTRY is undefined....tracking that down

**Comment by jstrachan** [comment]
BTW I fixed the `DOCKER_REGISTRY` issue yesterday. If you do `jx upgrade platform` and delete and recreate your devpod `jx delete devpod && jx create devpod` it should be fine again.

On the `watch.sh` really those scripts were temporary hacks until skaffold did all the awesome via `skaffold dev`; though we've only `watch.sh` scripts in the quickstarts for node + go right now:
* https://github.com/jenkins-x-quickstarts/node-http/blob/master/watch.sh
* https://github.com/jenkins-x-quickstarts/golang-http/blob/master/watch.sh

the aim is to move towards just using vanilla `skaffold` really. We're super close to moving back to the upstream skaffold now. Now the remaining known regression was fixed: https://github.com/GoogleContainerTools/skaffold/issues/741

we just need the next release and we're good to go.

BTW we're hoping to use a slightly different approach to give a nice developer feedback for working with Java/Spring: https://github.com/jenkins-x/jx/issues/833
